src/index.ts

```diff
@@ -96,6 +103,13 @@ const extension: JupyterFrontEndPlugin<void> = {
     let reportCellExecutionTime = true;
     let reportCellNumber = true;
     let cellNumberType = 'cell_index';
+    const cellExecutionMetadataTable: {
+      [cellId: string]: ICellExecutionMetadata;
+    } = {};
```

Comment:
When a user deletes a running cell before its completion, the scheduled items remain in the `cellExecutionMetadataTable`.
How about using the LRU cache for `cellExecutionMetadataTable` as well?

Reply:
mmh yeah that can happen. yep, an LRU cache makes sense here too. I'm thinking it maybe enough to allow storing 500 notebooks * 5 cells = 2,500 items at max?

Reply:
yeah, I agree.